### PR TITLE
add: cloudflared protocol option

### PR DESCRIPTION
**Pull request by omegazeng**
https://github.com/STRRL/cloudflare-tunnel-ingress-controller/issues/72
use QUIC by default

add `--set cloudflared.protocol=http2` to use HTTP2

### CODE REVIEW

**Comment by omegazeng** [comment]
Hi  @STRRL thank you for this amazing project.
the `cloudflared` has many flags, it may need another solution to support it.
for example:
```yaml
cloudflared:
  extra-args:
  - --edge-ip-version
  - "6"
  - --protocol
  - http2
```

**Comment by STRRL** [comment]
LGTM Thanks! 

BTW, about cloudflared customization, we're discussing in #128, feel free to leave comments if you have any other idea, or maybe hit 👍 if you think current proposal would work for you

File: helm/cloudflare-tunnel-ingress-controller/values.yaml [hunk 1]
```diff
@@ -77,3 +79,4 @@ cloudflared:
     pullPolicy: IfNotPresent
     tag: latest
   replicaCount: 1
+  protocol: quic
```
Comment (z0rc) [inline helm/cloudflare-tunnel-ingress-controller/values.yaml:82]: The default must be `auto`, not `quic`. `auto` is quic with fallback to `http2`.

Comment (STRRL) [reply helm/cloudflare-tunnel-ingress-controller/values.yaml:82]: 🤔 I would update it

Comment (omegazeng) [reply helm/cloudflare-tunnel-ingress-controller/values.yaml:82]: Sorry. I checked the Cloudflared docs, default is `auto`.
Weird. when I used the default config,  it did not fallback to `http2`, and the tunnel did not work. I thought the default value was `quic`.

Comment (STRRL) [reply helm/cloudflare-tunnel-ingress-controller/values.yaml:82]: default protocol `quic` -> `http2` updated in aa08f8b

Comment (STRRL) [reply helm/cloudflare-tunnel-ingress-controller/values.yaml:82]: > Sorry. I checked the Cloudflared docs, default is `auto`. Weird. when I used the default config, it did not fallback to `http2`, and the tunnel did not work. I thought the default value was `quic`.

just curious, does explicitly setting cloudflared protocol to `http2` work for you?

**Pull request merged**